### ISSUE TITLE
Inconsistent overwriting of headers

### ISSUE THREAD

**Issue by jgerken** [traefik/traefik#6723]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I tried to set up Træfik with some default headers via a default middleware in a convention-over-configuration fashion. The middleware defined in the entry point defines a standard set of security headers which can then be overwritten by defining middlewares for a router.

When only the default middleware is active everything is fine and the headers are reported as expected:

```
curl -sD - -o /dev/null  http://localhost

HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 612
Content-Type: text/html
Date: Thu, 23 Apr 2020 14:17:11 GMT
Etag: "5e95c66e-264"
Last-Modified: Tue, 14 Apr 2020 14:19:26 GMT
Referrer-Policy: strict-origin-when-cross-origin
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Test: foobar
X-Xss-Protection: 1; mode=block
```

When I add a middleware that only contains an additional header (`custom@file`, first commented line in the attached `docker-compose.yml`), the output is still fine:

```
curl -sD - -o /dev/null  http://localhost

HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 612
Content-Type: text/html
Date: Thu, 23 Apr 2020 14:18:44 GMT
Etag: "5e95c66e-264"
Last-Modified: Tue, 14 Apr 2020 14:19:26 GMT
Referrer-Policy: strict-origin-when-cross-origin
X-Content-Type-Options: nosniff
X-Custom-Response-Header: value
X-Frame-Options: DENY
X-Test: foobar
X-Xss-Protection: 1; mode=block
```

However, when I activated the middleware `allow-x-frame-self@file`(regardless whether it was used together with `custom@file` or not; second and third commented lines in the attached `docker-compose.yml`) the headers were no longer correct.

Without having inspected the source code, it seems for me like a second instance of the headers middleware is overwriting/resetting all values from the previous one expect for the custom headers. This might also be the root cause for #5568.

Note: for this bug report, I have reduced it to http and removed https configuration, forwarding and STS headers but that use case is also affected by this bug.

### What did you expect to see?

```
curl -sD - -o /dev/null  http://localhost

HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 612
Content-Type: text/html
Date: Thu, 23 Apr 2020 14:18:44 GMT
Etag: "5e95c66e-264"
Last-Modified: Tue, 14 Apr 2020 14:19:26 GMT
Referrer-Policy: strict-origin-when-cross-origin
X-Content-Type-Options: nosniff
X-Custom-Response-Header: value
X-Frame-Options: SAMEORIGIN
X-Test: foobar
X-Xss-Protection: 1; mode=block
```

### What did you see instead?

```
curl -sD - -o /dev/null  http://localhost

HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 612
Content-Type: text/html
Date: Thu, 23 Apr 2020 14:19:48 GMT
Etag: "5e95c66e-264"
Last-Modified: Tue, 14 Apr 2020 14:19:26 GMT
X-Frame-Options: SAMEORIGIN
X-Test: foobar
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.0
Codename:     chevrotin
Go version:   go1.14.1
Built:        2020-03-25T17:32:57Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

I have attached all my configuration files in one zip to this ticket.
[bug.zip](https://github.com/containous/traefik/files/4523160/bug.zip)

**Comment by reminec** [comment]
I confirm ! ;(

When you chain middlewares, for example 'security-headers@docker,myapp-headers' then value of 'security-headers' are ignored or reset.

You can switch order to confirm... (ie : 'myapp-headers,security-headers@docker')

**Comment by dtomcej** [comment]
This is related to https://github.com/containous/traefik/issues/5568, but also related to https://github.com/containous/traefik/issues/5568

Should be fixed by https://github.com/containous/traefik/pull/6604 and https://github.com/containous/traefik/pull/6750

**Comment by ldez** [comment]
Fixed by #7230
